### PR TITLE
New command: kovid top n

### LINKED ISSUE #111: Add a new command `kovid top N`
Add a new command that returns `top N` by deaths and cases.

### PR DESCRIPTION
**Pull request by volfgox**
Try to resolve #111 

Added a feature to list top countries or US states in two possible incidents, number of cases or deaths.

The MR needs screenshots for readme page to be ready to merge.

### CODE REVIEW

File: lib/kovid/constants.rb [hunk 1]
```diff
@@ -123,5 +123,17 @@ module Constants
       compare_us_states: [1, 2, 3, 4, 5],
       compare_provinces: [1, 2, 3]
     }.freeze
+
+    COUNTRY_CASES_DEATHS = [
```
Comment (siaw23) [inline lib/kovid/constants.rb:127]: Can include other information in this table so it looks similar to this:

```
+--------+-------------+--------+--------------+-----------+---------+
|                             🇵🇱​ POLAND                              |
+--------+-------------+--------+--------------+-----------+---------+
| Cases  | Cases Today | Deaths | Deaths Today | Recovered | Tests   |
+--------+-------------+--------+--------------+-----------+---------+
| 17,062 | +141        | 847    | +8           | 9,805     | 507,778 |
+--------+-------------+--------+--------------+-----------+---------+
```
Same with `STATE_CASES_DEATHS` :)

File: lib/kovid/tablelize.rb [hunk 2]
```diff
@@ -223,6 +239,31 @@ def align_columns(table_type, table)
...
+                         end
+        [
+          location_title,
+          Kovid.comma_delimit(data['cases']),
+          Kovid.comma_delimit(data['deaths'])
+        ]
+      end
+
+      def top_heading(options)
+        return COUNTRY_CASES_DEATHS if options[:location] == :countries
```
Comment (siaw23) [inline lib/kovid/tablelize.rb:257]: I'd love if you used a ternary operator here instead. What do you think? :)

**Comment by siaw23** [comment]
I think this is an excellent job! I have made some really small comments though. ❤️

File: README.md [hunk 3]
```diff
@@ -76,6 +76,12 @@ ___
 * `kovid history COUNTRY N` (history in the last N days).
 * `kovid history STATE --usa`
 ___
+😷 **Tops**
+* `kovid top N` (top N countries in number of cases).
+* `kovid top N --deaths` (top N countries in number of deaths).
```
Comment (siaw23) [inline README.md:81]: It'll be nice to know that `kovid top N -d` will work too here.

File: lib/kovid/constants.rb (hunk 1 above)
Comment (siaw23) [reply lib/kovid/constants.rb:127]: I'm talking about the headings(kind of info we want)*

Comment (volfgox) [reply lib/kovid/constants.rb:127]: It seems nice to have `--full` options too.

Comment (siaw23) [reply lib/kovid/constants.rb:127]: Adding `--full` will be too many commands/work. You can just fetch the full info for the table by default so there'll be no need for `--full`. Users can then just look and find what they want.

File: lib/kovid/tablelize.rb (hunk 2 above)
Comment (volfgox) [reply lib/kovid/tablelize.rb:257]: First, I used ternary operator. But, I got a long line. So, for sake of readability, I turned it to this.

File: lib/kovid/constants.rb (hunk 1 above)
Comment (volfgox) [reply lib/kovid/constants.rb:127]: Nice.

File: lib/kovid/tablelize.rb (hunk 2 above)
Comment (volfgox) [reply lib/kovid/tablelize.rb:257]: It becomes multi-line ternary operator. So, I prefer using `if` in a guard clause.

**Comment by siaw23** [comment]
Nice :). Thank you very much for your work. I'll push a new version in a few minutes with your changes!! <3

**Comment by volfgox** [comment]
Thank you very much.
As I mentioned in the PR body, we should have updated screenshots in the README before merge.

**Comment by volfgox** [comment]
@siaw23 should we create another issue for updating screenshots?
I didn't set screenshots. Because I couldn't create screenshots similar to previous ones. And this would have resulted in inconsistency in README.

**Comment by siaw23** [comment]
I have already fixed the screenshot. Please have a look at the README now. I just added the screenshots to your text :)

**Comment by siaw23** [comment]
Thanks a lot once again for the amazing work! :). A new version of `kovid` has now been released to the public! @volfgox

**Comment by volfgox** [comment]
It was a greet experiment for me. This was a my first open source contribution. So, I totally feel happy :)

**Comment by siaw23** [comment]
Awesome! I'm giving you credit on twitter. Do you have a Twitter account I can tag you in my tweet? :)

**Comment by volfgox** [comment]
It is great! I appreciate it. [@HazhirDerakhshi](https://twitter.com/HazhirDerakhshi)

**Comment by siaw23** [comment]
If it's not too much to ask. The 'IN CASES/DEATHS' in this image should say "BY CASES/DEATHS" :)

Screenshot 2020-05-13 at 15 41 20

@volfgox 

:)

**Comment by volfgox** [comment]
I see. I will fix this.

**Pull request merged**